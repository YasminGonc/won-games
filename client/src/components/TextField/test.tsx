import { renderWithTheme } from '../../utils/tests/helpers'
import { screen } from '@testing-library/react'

import { TextField } from '.'
import { UserCircle } from 'phosphor-react'

describe('<TextField />', () => {
  it('should render placeholder by default', () => {
    renderWithTheme(<TextField placeholder="Name" />)

    expect(screen.getByLabelText('Name input')).toBeInTheDocument()
    expect(screen.getByLabelText('Name input')).toHaveAttribute('type', 'text')
    expect(screen.getByPlaceholderText('Name')).toBeInTheDocument()
  })

  it('should render a label when it is passed', () => {
    renderWithTheme(<TextField placeholder="Name" label="Name" />)

    expect(screen.getByText(/name/i)).toBeInTheDocument()
  })

  it('should render with an icon at the left position', () => {
    renderWithTheme(
      <TextField
        placeholder="Name"
        label="Name"
        icon={<UserCircle data-testid="icon" />}
      />
    )

    expect(screen.getByTestId('icon')).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toHaveStyle('order: 0')
  })

  it('should render with an icon in the right position when it passed', () => {
    renderWithTheme(
      <TextField
        placeholder="Name"
        label="Name"
        icon={<UserCircle data-testid="icon" />}
        iconPosition="right"
      />
    )

    expect(screen.getByTestId('icon')).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toHaveStyle('order: 1')
  })

  it('should render disabled', () => {
    renderWithTheme(<TextField placeholder="Name" disabled />)

    expect(screen.getByRole('textbox')).toBeDisabled()
  })

  it('should render an error', () => {
    const { container } = renderWithTheme(
      <TextField placeholder="Name" label="Name" error="Something wrong" />
    )

    expect(screen.getByText('Something wrong')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
