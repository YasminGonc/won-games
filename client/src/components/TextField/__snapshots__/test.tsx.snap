// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render an error 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .c1 {
  color: #FF6347;
}

.c0 .c3 {
  box-shadow: 0 0 0.5rem #FF6347;
}

.c2 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
  margin-bottom: 0.8rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  background-color: #EAEAEA;
  border-radius: 0.4rem;
  padding: 0.8rem;
}

.c4:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c4 svg {
  height: 24px;
  width: 24px;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c5 {
  color: #2E2F42;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  background-color: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5::-webkit-input-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c5::-moz-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c5:-ms-input-placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c5::placeholder {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
}

.c6 {
  margin-top: 0.8rem;
  color: #FF6347;
  font-size: 1.2rem;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
    for="Name"
  >
    Name
  </label>
  <div
    class="c3 c4"
  >
    <input
      aria-label="Name input"
      class="c5"
      id="Name"
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
  <span
    class="c6"
  >
    Something wrong
  </span>
</div>
`;
